mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2285,7 +2284,9 @@ fn broadcast_keep_alive_for_pub(ctx: &MmArc, pubkey: &str, orderbook: &Orderbook
         timestamp: now_sec(),
     };
 
-    broadcast_ordermatch_message(ctx, topics, message.into(), p2p_privkey);
+    for topic in topics {
+        broadcast_ordermatch_message(ctx, topic, message.clone().into(), p2p_privkey);
+    }
```

Comment:
Should we broadcast all roots to each topic? I think if we should only broadcasted the root related to the pair/topic. I think we can even remove the below but I am not sure https://github.com/KomodoPlatform/komodo-defi-framework/blob/a78e6bcf6910f10ddb8463907661a136b9c7ef3b/mm2src/mm2_main/src/lp_ordermatch.rs#L2487

@artemii235 your opinion regarding this would be useful :)

Reply:
Oh, very true. Good catch, thanks! :)

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/1878/commits/2dc9bcfc659965151d630ab2c9c0df83a5fe2156 should do the work

Reply:
@shamardy `topics_subscribed_to` is used to track whether the node has subscribed to a topic and requested the *full* orderbook for the selected pair. 
https://github.com/KomodoPlatform/komodo-defi-framework/blob/4dbd3b88d438ebd8ce48896e6880916a08f8d8ef/mm2src/mm2_main/src/lp_ordermatch.rs#L5458-L5470
Theoretically, it can be removed, but we will then need to request topic subscription state from P2P layer to prevent excessive orderbook syncing requests. I would say that such refactoring is not needed 🙂